src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserAccessService.kt

```diff
@@ -192,13 +193,28 @@ class UserAccessService(
     else -> false
   }
 
+  /**
+   * This function only checks if the user has the correct permissions to withdraw the given application.
+   *
+   * It doesn't consider if the application is in a withdrawable state
+   */
   fun userCanWithdrawApplication(user: UserEntity, application: ApplicationEntity): Boolean = when (application) {
```

Comment:
just a tiny suggestion: would `userMayWithdrawApplication()` help convey that this is about permission rather than ability?

Reply:
Good idea. I'll do this on one of my subsequent PRs that will follow this as I have a lot stacked up so changing older ones is a bit painful!